sql/2020/19_Compression/19_02.byte_saving_of_text_compression.sql

```diff
@@ -0,0 +1,22 @@
+#standardSQL
+#Text Based Compression Byte Savings
+SELECT 
+  _TABLE_SUFFIX AS client,
+  if(JSON_EXTRACT_SCALAR(report, "$.audits.uses-text-compression.score") != "1", 'non_compression', 'compression') AS compression,
+  percents,
+  APPROX_QUANTILES(( CAST(JSON_EXTRACT_SCALAR(report, "$.audits.uses-text-compression.details.overallSavingsBytes") AS INT64) / 1024), 1000)[OFFSET(percents * 10)] AS kbyte_savings,
+  COUNT(0) AS num_pages,
+  SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS total,
+  COUNT(0) / SUM(COUNT(0)) OVER (PARTITION BY _TABLE_SUFFIX) AS pct_pages       
+FROM
+  `httparchive.lighthouse.2020_08_01_*`,
+  unnest([10,25,50,75,90]) AS percents
+WHERE
+  report IS NOT NULL AND 
+  JSON_EXTRACT_SCALAR(report, "$.audits.uses-text-compression.score")  IS NOT NULL
+GROUP BY 
+  _TABLE_SUFFIX, 
+  report,
```

Comment:
I think this is why the query is exceeding execution limits.
```suggestion
  compression,
```

Reply:
great thanks to the tips that fix it well.